cpp/src/copying/concatenate.cu

```diff
@@ -70,14 +70,18 @@ auto create_device_views(std::vector<column_view> const& views, cudaStream_t str
   auto d_views = rmm::device_vector<column_device_view>{device_views};
 
   // Compute the partition offsets
+  // Note: Using 64-bit size_t so we can detect overflow of 32-bit size_type
+  // Note: Using separate transform and inclusive_scan because transform_inclusive_scan fails to
+  // compile with:
+  // error: no suitable constructor exists to convert from "size_t" to "cudf::column_device_view"
```

Comment:
We should debug why we get this error first.

Reply:
Agree. We should not de-fuse these kernels if possible.

Reply:
As an aside, I was under the impression that nvcc had some facility to fuse multiple thrust calls like this into one effective kernel.  Is that not the case or is it just not as good as a proper hand-rolled one?

Reply:
nvcc cannot fuse independent kernels into a single kernel.

Reply:
I think I have a fix for this in Thrust. Will send out a PR soon.

Reply:
https://github.com/thrust/thrust/issues/1260

Reply:
Fixed.